### PR TITLE
Add conformance tests runner

### PR DESCRIPTION
**Pull request by menduz**
Uses zig build system to build the conformance test runner.

To use it:

```
cd conformance
zig build conformance-run
```

It will:

1) download protoc (sources), abseil and jsoncpp. then build the test harness binary
2) it will build the testee binary
3) run the tests

Results are shocking in terms of numbers, but not in terms of functionality. Most of them are JSON serialization errors which are not core to what we have been developing. It is interesting though to have the tests as guidance for new features.

~`CONFORMANCE SUITE FAILED: 813 successes, 4091 skipped, 0 expected failures, 663 unexpected failures.`~

CONFORMANCE SUITE PASSED: 1476 successes, 4091 skipped, 0 expected failures, 0 unexpected failures.

---

Updated, all tests passing. It turned out to be almost 99% JSON (de)serialization and custom behavior of builtin types in the namespace `google.protobuf.Timestamp`. Those were purposefully left unimplemented in the past as they are not recommended by google anymore. But facing a benchmark saying that the library is not compliant, it makes sense to update them.

The test suite also found a real panic error, and some minor serialization corner cases i.e. to interpret booleans. The official spec is not in sync with the conformance tests for that one, I've picked the conformance tests path for the sake of consistency.

### CODE REVIEW

File: src/wire.zig [hunk 1]
```diff
@@ -382,9 +382,6 @@ pub fn decodeRepeated(
     ).pointer.child;
     comptime std.debug.assert(ResultList == std.ArrayList(Result));
 
-    const current_capacity = result.capacity;
-    errdefer result.shrinkAndFree(allocator, current_capacity);
-
```
Comment (menduz) [inline src/wire.zig:387]: This was causing a real panic in the conformance tests!

**Comment by menduz** [comment]
CONFORMANCE TEST BEGIN ====================================

CONFORMANCE SUITE PASSED: 1476 successes, 4091 skipped, 0 expected failures, 0 unexpected failures.

WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1778708159.993881 2275666 fork_pipe_runner.cc:177] ./.zig-cache/o/b5c1909c6125b166e6e8abcd8694dc10/conformance-testee
CONFORMANCE TEST BEGIN ====================================

CONFORMANCE SUITE PASSED: 0 successes, 883 skipped, 0 expected failures, 0 unexpected failures.

info: conformance-zig-protobuf: served 883 tests
info: conformance-zig-protobuf: served 5567 tests

**Comment by Arwalk** [comment]
@menduz ~~I feel like there is way too much stuff in this PR, can we break it down? I'm ok with a stacked PR approach.~~

~~I think we should have a PR for just building and running the conformance tests, with the minimum required for it, then a PR for each fixed test (or group of, if multiple tests are fixed by a single change).~~

~~Also if we could focus on the wire, and do json later ?~~

EDIT: didn't realize you edited and made it mostly pass.

Can we at least have 3 commits:

- making the build for conformance tests
- fixing wire tests
- fixing json tests

also do we need to keep the  `conformance/out/failing_tests.txt` file ?

**Comment by menduz** [comment]
Superseded by #170 and #171
